api/internal/handler/consumer/consumer_test.go

```diff
@@ -18,230 +18,381 @@
+		getCalled := true
+		mStore := &store.MockInterface{}
+		mStore.On("List", mock.Anything).Run(func(args mock.Arguments) {
+			getCalled = true
+			input := args.Get(0).(store.ListInput)
+			assert.Equal(t, tc.wantInput.PageSize, input.PageSize, tc.caseDesc)
+			assert.Equal(t, tc.wantInput.PageNumber, input.PageNumber, tc.caseDesc)
+		}).Return(func(input store.ListInput) *store.ListOutput {
+			var returnData []interface{}
+			for _, c := range tc.giveData {
+				if input.Predicate(c) {
+					returnData = append(returnData, c)
+				}
+			}
+			return &store.ListOutput{
+				Rows:      returnData,
+				TotalSize: len(returnData),
+			}
+		}, tc.giveErr)
 
+		h := Handler{consumerStore: mStore}
+		ctx := droplet.NewContext()
+		ctx.SetInput(tc.giveInput)
+		ret, err := h.List(ctx)
+		assert.True(t, getCalled, tc.caseDesc)
+		assert.Equal(t, tc.wantRet, ret, tc.caseDesc)
+		assert.Equal(t, tc.wantErr, err, tc.caseDesc)
+	}
+}
+
+func TestHandler_Create(t *testing.T) {
+	tests := []struct {
+		caseDesc   string
+		giveInput  *entity.Consumer
+		giveCtx    context.Context
+		giveErr    error
+		wantErr    error
+		wantInput  *entity.Consumer
+		wantRet    interface{}
+		wantCalled bool
+	}{
+		{
+			caseDesc: "normal",
+			giveInput: &entity.Consumer{
+				Username: "name",
+				Plugins: map[string]interface{}{
+					"jwt-auth": map[string]interface{}{},
+				},
+			},
+			giveCtx: context.WithValue(context.Background(), "test", "value"),
+			wantInput: &entity.Consumer{
+				BaseInfo: entity.BaseInfo{
+					ID: "name",
+				},
+				Username: "name",
+				Plugins: map[string]interface{}{
+					"jwt-auth": map[string]interface{}{
+						"exp": 86400,
+					},
+				},
+			},
+			wantRet:    nil,
+			wantCalled: true,
+		},
+		{
+			caseDesc: "create failed",
+			giveInput: &entity.Consumer{
+				Username: "name",
+				Plugins: map[string]interface{}{
+					"jwt-auth": map[string]interface{}{
+						"exp": 5000,
+					},
+				},
+			},
+			giveErr: fmt.Errorf("create failed"),
+			wantInput: &entity.Consumer{
+				BaseInfo: entity.BaseInfo{
+					ID: "name",
+				},
+				Username: "name",
+				Plugins: map[string]interface{}{
+					"jwt-auth": map[string]interface{}{
+						"exp": 5000,
+					},
+				},
+			},
+			wantErr: fmt.Errorf("create failed"),
+			wantRet: &data.SpecCodeResponse{
+				StatusCode: http.StatusInternalServerError,
+			},
+			wantCalled: true,
+		},
+	}
+
+	for _, tc := range tests {
+		methodCalled := true
+		mStore := &store.MockInterface{}
+		mStore.On("Create", mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
+			methodCalled = true
+			assert.Equal(t, tc.giveCtx, args.Get(0), tc.caseDesc)
+			assert.Equal(t, tc.wantInput, args.Get(1), tc.caseDesc)
+		}).Return(tc.giveErr)
+
+		h := Handler{consumerStore: mStore}
+		ctx := droplet.NewContext()
+		ctx.SetInput(tc.giveInput)
+		ctx.SetContext(tc.giveCtx)
+		ret, err := h.Create(ctx)
+		assert.Equal(t, tc.wantCalled, methodCalled, tc.caseDesc)
```

Comment:
seems `wantCalled` is aways true ?

Reply:
In this case: Yes.
Because the handler function has no logic can stop it to call `store.Create`.